docker/runtime/php-7.2/http-trigger/Controller.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+namespace Kubeless;
+
+use \Psr\Http\Message\ServerRequestInterface as Request;
+use \Psr\Http\Message\ResponseInterface as Response;
+
+require 'vendor/autoload.php';
+
+class TimeoutFunctionException extends \RuntimeException {}
+
+class Controller
+{
+  private $app;
+  private $timeout;
+  private $root;
+  private $function;
+  private $currentDir;
+
+  public function __construct()
+  {
+
+    $this->app = new \Slim\App();
+    $this->timeout = (!empty(getenv('FUNC_TIMEOUT')) ? getenv('FUNC_TIMEOUT') : 180);
+    $this->root = (!empty(getenv('MOD_ROOT_PATH')) ? getenv('MOD_ROOT_PATH') : '/kubeless/');
+    $this->file = sprintf("/kubeless/%s.php", getenv('MOD_NAME'));
+    $this->function = getenv('FUNC_HANDLER');
+    $this->currentDir = getcwd();
+  }
+
+  /**
+   * Run the injected function.
+   *
+   * @return void
+   */
+  private function runFunction(Request $request)
+  {
+      set_time_limit($this->timeout);
+      ob_start();
+      chdir($this->root);
+      include $this->file;
+      if (!function_exists($this->function)) {
+        throw new \Exception(sprintf("Function %s not exist", $this->function));
+      }
+      $pid = pcntl_fork();
+      if ($pid == 0) {
+        call_user_func_array($this->function, [$request]);
+        $response = ob_get_contents();
+        ob_end_clean();
+        chdir($this->currentDir);
+
+        return $response;
+      } else {
+          sleep($this->timeout);
+          posix_kill($pid, SIGKILL);
+          throw new TimeoutFunctionException();
```

Comment:
does this throw the exception even if the process were already finished at the time of `posix_kill`?

Reply:
No the request dies when parent or child return a response